Source: Arch Linux cairo-ubuntu package (refreshed for cairo 1.12.2)
$ wget http://aur.archlinux.org/packages/cairo-ubuntu/cairo-ubuntu.tar.gz

--- a/src/cairo-ft-font.c
+++ b/src/cairo-ft-font.c
@@ -1766,8 +1766,15 @@
     if (options->base.hint_style == CAIRO_HINT_STYLE_DEFAULT)
 	options->base.hint_style = other->base.hint_style;
 
-    if (other->base.hint_style == CAIRO_HINT_STYLE_NONE)
-	options->base.hint_style = CAIRO_HINT_STYLE_NONE;
+    //if (other->base.hint_style == CAIRO_HINT_STYLE_NONE)
+	//options->base.hint_style = CAIRO_HINT_STYLE_NONE;
+
+	if (other->base.hint_style == CAIRO_HINT_STYLE_NONE ||
+		other->base.hint_style == CAIRO_HINT_STYLE_SLIGHT ||
+		other->base.hint_style == CAIRO_HINT_STYLE_MEDIUM ||
+		other->base.hint_style == CAIRO_HINT_STYLE_FULL) {
+	options->base.hint_style = other->base.hint_style;
+	}
 
     if (options->base.lcd_filter == CAIRO_LCD_FILTER_DEFAULT)
 	options->base.lcd_filter = other->base.lcd_filter;
